Fails to build on s390x: #include <s390utils/vtoc.h>

libblock 2.12
Debian unstable

libblockdev fails to build on s390x.

https://github.com/storaged-project/libblockdev/blob/master/src/lib/plugin_apis/s390.api#L7

Where is `s390utils/vtoc.h` ?

[Full build log](https://buildd.debian.org/status/fetch.php?pkg=libblockdev&arch=s390x&ver=2.12-2&stamp=1504190590)

Build log excerpt
---------------------
```
make[4]: Leaving directory '/<<PKGBUILDDIR>>/src/lib/plugin_apis'
make[4]: Entering directory '/<<PKGBUILDDIR>>/src/lib'
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.12\" -DPACKAGE_STRING=\"libblockdev\ 2.12\" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.12\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 -DWITH_BD_BCACHE=/\*\*/ -DWITH_BD_BTRFS=/\*\*/ -DWITH_BD_CRYPTO=/\*\*/ -DWITH_BD_DM=/\*\*/ -DWITH_BD_LOOP=/\*\*/ -DWITH_BD_LVM=/\*\*/ -DWITH_BD_LVM_DBUS=/\*\*/ -DWITH_BD_MDRAID=/\*\*/ -DWITH_BD_MPATH=/\*\*/ -DWITH_BD_SWAP=/\*\*/ -DWITH_BD_KBD=/\*\*/ -DWITH_BD_PART=/\*\*/ -DWITH_BD_FS=/\*\*/ -I.  -I./../../include/ -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=gnu99 -c -o libblockdev_la-blockdev.lo `test -f 'blockdev.c' || echo './'`blockdev.c
/bin/bash ../../libtool  --tag=CC   --mode=compile gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.12\" -DPACKAGE_STRING=\"libblockdev\ 2.12\" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.12\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 -DWITH_BD_BCACHE=/\*\*/ -DWITH_BD_BTRFS=/\*\*/ -DWITH_BD_CRYPTO=/\*\*/ -DWITH_BD_DM=/\*\*/ -DWITH_BD_LOOP=/\*\*/ -DWITH_BD_LVM=/\*\*/ -DWITH_BD_LVM_DBUS=/\*\*/ -DWITH_BD_MDRAID=/\*\*/ -DWITH_BD_MPATH=/\*\*/ -DWITH_BD_SWAP=/\*\*/ -DWITH_BD_KBD=/\*\*/ -DWITH_BD_PART=/\*\*/ -DWITH_BD_FS=/\*\*/ -I.  -I./../../include/ -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=gnu99 -c -o libblockdev_la-plugins.lo `test -f 'plugins.c' || echo './'`plugins.c
libtool: compile:  gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.12\" "-DPACKAGE_STRING=\"libblockdev 2.12\"" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.12\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 "-DWITH_BD_BCACHE=/**/" "-DWITH_BD_BTRFS=/**/" "-DWITH_BD_CRYPTO=/**/" "-DWITH_BD_DM=/**/" "-DWITH_BD_LOOP=/**/" "-DWITH_BD_LVM=/**/" "-DWITH_BD_LVM_DBUS=/**/" "-DWITH_BD_MDRAID=/**/" "-DWITH_BD_MPATH=/**/" "-DWITH_BD_SWAP=/**/" "-DWITH_BD_KBD=/**/" "-DWITH_BD_PART=/**/" "-DWITH_BD_FS=/**/" -I. -I./../../include/ -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=gnu99 -c blockdev.c  -fPIC -DPIC -o .libs/libblockdev_la-blockdev.o
libtool: compile:  gcc -DPACKAGE_NAME=\"libblockdev\" -DPACKAGE_TARNAME=\"libblockdev\" -DPACKAGE_VERSION=\"2.12\" "-DPACKAGE_STRING=\"libblockdev 2.12\"" -DPACKAGE_BUGREPORT=\"vpodzime@redhat.com\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libblockdev\" -DVERSION=\"2.12\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DHAVE_DLFCN_H=1 -DHAVE_STRING_H=1 -DHAVE_UNISTD_H=1 -DHAVE_SYS_FCNTL_H=1 -DHAVE_SYS_IOCTL_H=1 -DHAVE_LINUX_RANDOM_H=1 -DHAVE_GLOB_H=1 -DHAVE_SYSLOG_H=1 -DHAVE_MATH_H=1 "-DWITH_BD_BCACHE=/**/" "-DWITH_BD_BTRFS=/**/" "-DWITH_BD_CRYPTO=/**/" "-DWITH_BD_DM=/**/" "-DWITH_BD_LOOP=/**/" "-DWITH_BD_LVM=/**/" "-DWITH_BD_LVM_DBUS=/**/" "-DWITH_BD_MDRAID=/**/" "-DWITH_BD_MPATH=/**/" "-DWITH_BD_SWAP=/**/" "-DWITH_BD_KBD=/**/" "-DWITH_BD_PART=/**/" "-DWITH_BD_FS=/**/" -I. -I./../../include/ -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -I/usr/include/glib-2.0 -I/usr/lib/s390x-linux-gnu/glib-2.0/include -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -std=gnu99 -c plugins.c  -fPIC -DPIC -o .libs/libblockdev_la-plugins.o
In file included from blockdev.c:30:0:
plugin_apis/s390.h:7:10: fatal error: s390utils/vtoc.h: No such file or directory
 #include <s390utils/vtoc.h>
```

--- a/configure.ac
+++ b/configure.ac
@@ -44,8 +44,11 @@ AS_IF([test "x$found_introspection" = xyes],
       [AC_SUBST(WITH_GI, 1)],
       [AC_SUBST(WITH_GI, 0)])
 
-uname -p|grep s390
-on_s390=$?
+AC_CANONICAL_HOST
+case $host_cpu in
+  s390|s390x) on_s390=yes ;;
+  *) on_s390=no ;;
+esac
 
 skip_patterns=""
 
@@ -56,10 +59,10 @@ AC_ARG_WITH([s390],
 
 AC_SUBST(WITH_S390, 0)
 if test "x$with_s390" != "xno"; then
-    AS_IF([test "$on_s390" = "1" -a "x$with_s390" = "xyes"],
+    AS_IF([test "x$on_s390" = "xno" -a "x$with_s390" = "xyes"],
            [LIBBLOCKDEV_SOFT_FAILURE([s390 support requested, but not on s390 arch])
             AM_CONDITIONAL(WITH_S390, false)],
-          [test "$on_s390" == "0" -a "x$with_s390" != "xno"],
+          [test "x$on_s390" == "xyes" -a "x$with_s390" != "xno"],
            [AC_SUBST(WITH_S390, 1)
             AM_CONDITIONAL(WITH_S390, true)
             LIBBLOCKDEV_CHECK_HEADER([s390utils/vtoc.h], [], [s390utils/vtoc.h not available])

--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,6 @@ if test "x$with_s390" != "xno"; then
           [test "x$on_s390" == "xyes" -a "x$with_s390" != "xno"],
            [AC_SUBST(WITH_S390, 1)
             AM_CONDITIONAL(WITH_S390, true)
-            LIBBLOCKDEV_CHECK_HEADER([s390utils/vtoc.h], [], [s390utils/vtoc.h not available])
            ],
           [AC_SUBST(WITH_S390, 0)
            AM_CONDITIONAL(WITH_S390, false)

--- a/configure.ac
+++ b/configure.ac
@@ -47,9 +47,29 @@ AS_IF([test "x$found_introspection" = xyes],
 uname -p|grep s390
 on_s390=$?
 
-AM_CONDITIONAL(ON_S390, test "$on_s390" = "0")
-
 skip_patterns=""
+
+AC_ARG_WITH([s390],
+    AS_HELP_STRING([--with-s390], [support s390 @<:@default=check@:>@]),
+    [],
+    [with_s390=check])
+
+AC_SUBST(WITH_S390, 0)
+if test "x$with_s390" != "xno"; then
+    AS_IF([test "$on_s390" = "1" -a "x$with_s390" = "xyes"],
+           [LIBBLOCKDEV_SOFT_FAILURE([s390 support requested, but not on s390 arch])
+            AM_CONDITIONAL(WITH_S390, false)],
+          [test "$on_s390" == "0" -a "x$with_s390" != "xno"],
+           [AC_SUBST(WITH_S390, 1)
+            AM_CONDITIONAL(WITH_S390, true)
+            LIBBLOCKDEV_CHECK_HEADER([s390utils/vtoc.h], [], [s390utils/vtoc.h not available])
+           ],
+          [AC_SUBST(WITH_S390, 0)
+           AM_CONDITIONAL(WITH_S390, false)
+          ])
+else
+  AM_CONDITIONAL(WITH_S390, false)
+fi
 
 AC_ARG_WITH([python3],
     AS_HELP_STRING([--with-python3], [support python3 @<:@default=check@:>@]),

--- a/src/lib/Makefile.am
+++ b/src/lib/Makefile.am
@@ -35,7 +35,7 @@ GIHEADERS = ${builddir}/plugin_apis/kbd.h \
 GIHEADERS += $(wildcard ${srcdir}/../utils/*.[ch])
 GIHEADERS += blockdev.c blockdev.h plugins.c plugins.h
 
-if ON_S390
+if WITH_S390
 GIHEADERS += ${builddir}/plugin_apis/s390.h
 endif
 

--- a/src/lib/plugin_apis/s390.api
+++ b/src/lib/plugin_apis/s390.api
@@ -1,10 +1,5 @@
 #include <glib.h>
-#include <linux/fs.h>
-#include <stdio.h>
-#include <string.h>
 #include <blockdev/utils.h>
-#include <asm/dasd.h>
-#include <s390utils/vtoc.h>
 
 GQuark bd_s390_error_quark (void) {
     return g_quark_from_static_string ("g-bd-s390-error-quark");

--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -53,7 +53,7 @@ lib_LTLIBRARIES += libbd_fs.la
 endif
 
 
-if ON_S390
+if WITH_S390
 lib_LTLIBRARIES += libbd_s390.la
 endif
 
@@ -139,7 +139,7 @@ libbd_kbd_la_CPPFLAGS = -I${builddir}/../../include/
 libbd_kbd_la_SOURCES = kbd.c kbd.h
 endif
 
-if ON_S390
+if WITH_S390
 libbd_s390_la_CFLAGS = $(GLIB_CFLAGS) -Wall -Wextra -Werror
 libbd_s390_la_LIBADD = $(GLIB_LIBS) ${builddir}/../utils/libbd_utils.la
 libbd_s390_la_LDFLAGS = -L${srcdir}/../utils/ -version-info 2:0:0 -Wl,--no-undefined
@@ -221,7 +221,7 @@ endif
 
 noinst_HEADERS = part_err.h
 
-if ON_S390
+if WITH_S390
 libinclude_HEADERS += s390.h
 endif
 

--- a/src/plugins/s390.c
+++ b/src/plugins/s390.c
@@ -447,7 +447,7 @@ gchar* bd_s390_zfcp_sanitize_lun_input (const gchar *lun, GError **error) {
     gchar *append = NULL;
 
     /* first make sure we're not being played */
-    if ((lun == NULL) || (!*lun) || (strlen(lun) > 16)) {
+    if ((lun == NULL) || (!*lun) || (strlen(lun) > 18)) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "LUN not specified or invalid");
         return NULL;

--- a/src/plugins/s390.c
+++ b/src/plugins/s390.c
@@ -24,7 +24,9 @@
 #include <string.h>
 #include <blockdev/utils.h>
 #include <asm/dasd.h>
-#include <s390utils/vtoc.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
 
 #include "s390.h"
 

--- a/src/plugins/s390.c
+++ b/src/plugins/s390.c
@@ -127,7 +127,6 @@ gboolean bd_s390_dasd_needs_format (const gchar *dasd, GError **error) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "Error checking status of device %s; device may not exist,"
                      " or status can not be read.", dasd);
-        g_clear_error (error);
         return FALSE;
     }
 
@@ -139,7 +138,6 @@ gboolean bd_s390_dasd_needs_format (const gchar *dasd, GError **error) {
     if (!rc) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "Error checking status of device %s.", dasd);
-        g_clear_error (error);
         return FALSE;
     }
 
@@ -368,7 +366,6 @@ gchar* bd_s390_sanitize_dev_input (const gchar *dev, GError **error) {
     if ((dev == NULL) || (!*dev)) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "Device number not specified or invalid");
-        g_clear_error (error);
         return NULL;
     }
 
@@ -418,7 +415,6 @@ gchar* bd_s390_zfcp_sanitize_wwpn_input (const gchar *wwpn, GError **error) {
     if ((wwpn == NULL) || (!*wwpn) || (strlen(wwpn) < 2)) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "WWPN not specified or invalid");
-        g_clear_error (error);
         return NULL;
     }
 
@@ -454,7 +450,6 @@ gchar* bd_s390_zfcp_sanitize_lun_input (const gchar *lun, GError **error) {
     if ((lun == NULL) || (!*lun) || (strlen(lun) > 16)) {
         g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
                      "LUN not specified or invalid");
-        g_clear_error (error);
         return NULL;
     }
 

--- a/src/plugins/s390.h
+++ b/src/plugins/s390.h
@@ -1,11 +1,5 @@
 #include <glib.h>
-#include <linux/fs.h>
-#include <stdio.h>
-#include <string.h>
 #include <blockdev/utils.h>
-#include <asm/dasd.h>
-#include <s390utils/vtoc.h>
-
 
 GQuark bd_s390_error_quark (void);
 #define BD_S390_ERROR bd_s390_error_quark ()

--- a/tests/s390_test.py
+++ b/tests/s390_test.py
@@ -21,61 +21,61 @@ class S390TestCase(unittest.TestCase):
     def test_device_input(self):
         """Verify that s390_sanitize_dev_input works as expected"""
         dev = "1234"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), '0.0.' + dev)
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), '0.0.' + dev)
 
         dev = "123456"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), '0.0.' + dev)
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), '0.0.' + dev)
 
         # the device number is padded on the left with 0s up to 4 digits
         dev = "123.abc"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), "0.0.0abc")
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), "0.0.0abc")
         dev = "abc"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), "0.0.0abc")
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), "0.0.0abc")
         dev = ".abc"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), "0.0.0abc")
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), "0.0.0abc")
 
         # a complete number is unchanged
         dev = "0.0.abcd"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), dev)
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), dev)
 
         # a too long number doesn't mean a crash no matter if it makes sense
         dev = "0.0.abcdefgh"
-        self.assertEqual(BlockDev.s390.sanitize_dev_input(dev), dev)
+        self.assertEqual(BlockDev.s390_sanitize_dev_input(dev), dev)
 
     def test_wwpn_input(self):
         """Verify that s390_zfcp_sanitize_wwpn_input works as expected"""
         # missing "0x" from beginning of wwpn; this should be added by fx
         wwpn = "01234567abcdefab"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_wwpn_input(wwpn), "0x01234567abcdefab")
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_wwpn_input(wwpn), "0x01234567abcdefab")
         # this should be fine as-is
         wwpn = "0x01234567abcdefab"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_wwpn_input(wwpn), wwpn)
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_wwpn_input(wwpn), wwpn)
 
         # too short
         wwpn = "a"
         with self.assertRaises(GLib.GError):
-            BlockDev.s390.zfcp_sanitize_wwpn_input(wwpn)
+            BlockDev.s390_zfcp_sanitize_wwpn_input(wwpn)
 
     def test_lun_input(self):
         """Verify that s390_zfcp_sanitize_lun_input works as expected"""
         # user does not prepend lun with "0x"; this should get added
         lun = "01234567abcdefab"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_lun_input(lun), "0x01234567abcdefab")
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_lun_input(lun), "0x01234567abcdefab")
         # a user enters a lun that is between 0 and 16 chars long (non-inclusive); 0 padding should be added to expand to 16
         lun = "0x123"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_lun_input(lun), "0x0123000000000000")
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_lun_input(lun), "0x0123000000000000")
         lun = "0x12345"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_lun_input(lun), "0x1234500000000000")
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_lun_input(lun), "0x1234500000000000")
         lun = "0x123456"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_lun_input(lun), "0x1234560000000000")
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_lun_input(lun), "0x1234560000000000")
         # this should be fine as-is
         lun = "0x1234567800000000"
-        self.assertEqual(BlockDev.s390.zfcp_sanitize_lun_input(lun), lun)
+        self.assertEqual(BlockDev.s390_zfcp_sanitize_lun_input(lun), lun)
 
         # too long
         lun = "12345678901234567890"
         with self.assertRaises(GLib.GError):
-            BlockDev.s390.zfcp_sanitize_lun_input(lun)
+            BlockDev.s390_zfcp_sanitize_lun_input(lun)
 
 
 @unittest.skipUnless(os.uname()[4].startswith('s390'), "s390x architecture required")

